W-15435765 Update CIM documentation (#117)
mulesoft/docs-accelerators: retail/3.2/modules/ROOT/pages/rcg-sfdc-customization-guide.adoc

=== Create Custom Object via Apex Script

> The`CustomObjectUtility` and `MetadataService` Apex classes are used to create a custom object programmatically. The source can be found on https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project and https://github.com/financialforcedev/apex-mdapi/blob/master/apex-mdapi/src/classes/MetadataService.cls[MetadataService] respectively. Download the project and follow the steps below.

Reviewer: ```suggestion
Use the`CustomObjectUtility` and `MetadataService` Apex classes to create a custom object programmatically. The source is on the https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project and https://github.com/financialforcedev/apex-mdapi/blob/master/apex-mdapi/src/classes/MetadataService.cls[MetadataService] respectively. Download the project and follow these steps:
```

Author: This is one of those changes that needs to be back-ported to beta.

=== Create Fields via Apex Script

> Use the Apex class `CustomFieldUtility` to create custom fields programmatically. See the https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project to find the source for this class. Once you have downloaded the project, follow these steps:

Reviewer: ```suggestion
Use the Apex class `CustomFieldUtility` to create custom fields programmatically. See the https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project to find the source for this class. Download the project and follow these steps:
```

Author: This is one of those changes that needs to be back-ported to beta.

== Create Platform Events and Apex Triggers

> The following Platform Event Object and Apex Triggers must be created in order to capture updates from Salesforce. The scripts to create these Apex Triggers and add fields to Platform Event Object are located in the https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project.

Reviewer: ```suggestion
To capture updates from Salesforce, create the following Platform Event Object and Apex Triggers. The scripts to create the Apex Triggers and add fields to the Platform Event Object are in the https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project.
```

. Click the *New Platform Event* button to create an Object. Complete the *Platform Information Details* fields, including *Label*, *Plural Label*, *Object with Salesforce Object*, and *Salesforce Platform Event API Name* as indicated in the previous table.
. Click *Save*.

> . To add Custom Fields & RelationShips to Platform Event Definitions, run the utility script `CreatePlatformEventsCustomFields.txt` available in the `rcg-common-resources/salesforce` folder of https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project.

Reviewer: ```suggestion
. To add Custom Fields & RelationShips to Platform Event Definitions, run the utility script `CreatePlatformEventsCustomFields.txt`, which is in the `rcg-common-resources/salesforce` folder of the https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project.
```

Author: This is one of those changes that needs to be back-ported to beta.

. Search for the Salesforce object. This opens a page that shows the details of the object.
. Click *Triggers* in the Details section to view a list of available triggers for the object.
. Click *New*.

> . Search for the Apex Trigger name, which is indicated in the previous table. Copy the Trigger Script code `CreateApexTriggers.txt` available in the `rcg-common-resources/salesforce` folder of https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project and paste it into the script editor.

Reviewer: ```suggestion
. Search for the Apex Trigger name, which is indicated in the previous table. Copy the Trigger Script code `CreateApexTriggers.txt` from the `rcg-common-resources/salesforce` folder of the https://anypoint.mulesoft.com/exchange/org.mule.examples/rcg-common-resources-src/[RCG Common Resources] project and paste it into the script editor.
```

Author: This is one of those changes that needs to be back-ported to beta.
